### PR TITLE
rpc: extend param unmarshaling

### PR DESCRIPTION
**Pull request by fyrchik**
This PR does 3 things:
- adds array parameter unmarshalling
- extend `Param` with convenient methods
- refactor tests into using tables to make it easier add new tests

### CODE REVIEW

File: pkg/rpc/param_test.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestParam_UnmarshalJSON(t *testing.T) {
+	msg := `["kek", 123, ["lul", 3]]`
```
Comment (volekerb) [inline pkg/rpc/param_test.go]: kek lul.. maaan! =) probably you can think of smth less fun?=)

File: pkg/rpc/server.go [hunk 2]
```diff
@@ -322,21 +322,22 @@ func (s *Server) getAccountState(reqParams Params, unspents bool) (interface{},
...
 	}
+
 	vm, _ := s.chain.GetTestVM()
 	vm.LoadScript(script)
 	_ = vm.Run()
 	result := &wrappers.InvokeResult{
 		State:       vm.State(),
 		GasConsumed: "0.1",
-		Script:      hexScript.GetString(),
+		Script:      reqParams[0].GetString(),
```
Comment (volekerb) [inline pkg/rpc/server.go]: so we're taking just the first param only?

File: pkg/rpc/server.go [hunk 3]
```diff
@@ -346,11 +347,10 @@ func (s *Server) sendrawtransaction(reqParams Params) (interface{}, error) {
 	var resultsErr error
 	var results interface{}
 
-	param, err := reqParams.ValueWithType(0, stringT)
-	if err != nil {
-		resultsErr = err
-	} else if byteTx, err := hex.DecodeString(param.GetString()); err != nil {
-		resultsErr = errInvalidParams
+	if len(reqParams) < 1 {
```
Comment (volekerb) [inline pkg/rpc/server.go]: why not to combine both conditions?

File: pkg/rpc/server_test.go [hunk 4]
```diff
@@ -7,141 +7,223 @@ import (
...
 
+	"github.com/CityOfZion/neo-go/pkg/core"
+	"github.com/CityOfZion/neo-go/pkg/rpc/wrappers"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
-func TestRPC(t *testing.T) {
-	chain, handler := initServerWithInMemoryChain(t)
+type executor struct {
```
Comment (volekerb) [inline pkg/rpc/server_test.go]: I really like how tests look like now. Thank you! cool!

File: pkg/rpc/param_test.go (hunk 1 above)
Comment (fyrchik) [reply pkg/rpc/param_test.go]: I will rename to "elem1" and "elem2". "kek" and "lul" are defaults which came to my mind first ;)

File: pkg/rpc/server.go (hunk 2 above)
Comment (roman-khimov) [reply pkg/rpc/server.go]: Yep, all the other ones are irrelevant for the invoker.

Comment (roman-khimov) [reply pkg/rpc/server.go]: We could probably check for these excessive parameters and fail, but that's a different story.

**Pull request merged**